src/librustdoc/html/render/cache.rs

```diff
@@ -251,6 +252,23 @@ fn get_index_type_name(clean_type: &clean::Type, accept_generic: bool) -> Option
     }
 }
 
+// Return a list of generic parameters for use in the search index.
+//
+// This function replaces bounds with types, so that `T where T: Debug` just becomes `Debug`.
+// It does return duplicates, and that's intentional, since search queries like `Result<usize, usize>`
+// are supposed to match only results where both parameters are `usize`.
```

Comment:
Please turn this comment into a doc comment (very useful when documenting private APIs).